src/conn.rs

```diff
@@ -620,14 +620,15 @@ impl<const N: usize, P: ConnectionPeer> Connection<N, P> {
                         self.state = State::Closed { err: Some(err) };
                     } else {
                         let seq = *syn;
-                        *attempts += 1;
+                        let new_attempts = *attempts + 1;
+                        *attempts = new_attempts;
                         let packet = self.syn_packet(seq);
                         let _ = self.socket_events.send(SocketEvent::Outgoing((
                             packet.clone(),
                             self.cid.peer.clone(),
                         )));
-                        self.unacked
-                            .insert_at(seq, packet, self.config.initial_timeout);
+                        let timeout = self.config.initial_timeout * 2u32.pow(new_attempts as u32);
```

Comment:
do we need to consider how max conn attempts and this backoff might exceed the idle timeout threshold? maybe it's not an issue. if it is, I suppose we could just put a condition on the idle timeout to only activate if we have moved pass the connecting state. I'm okay if we leave it alone, but I thought it was worth calling out.

Reply:
Yeah, it's a good thought! We already choose the defaults to avoid the situation, but need to handle the custom configurations.

So what should utp_rs do if someone configures the max attempts so that they would keep trying past the idle timeout?

Option 1: The default behavior here is to just time out anyway, and you never get to try as many connection reattempts as you configured (because the other value you configured was too small). I don't think this is terrible, but should be documented on the Config struct.

Option 2: We ignore the idle timeout until the connection is established. This is probably about as surprising as Option 1, so similarly would require configuration docstrings.

Option 3: When connecting, reject configurations as invalid, if they have internal conflicts

Since Option 1 and 2 both have surprises that require documentation, I'm inclined to go with the one that requires less code: Option 1.